Dockerfile

```diff
@@ -1,21 +1,59 @@
-ARG BASE_IMAGE=nvidia/cuda:11.7.1-devel-ubuntu22.04
+ARG BASE_IMAGE=base
+# Stage from full image tag name for dependabot detection
+FROM nvidia/cuda:11.7.1-devel-ubuntu22.04 as base
+
+################################################################################
+# MARK: prepper - prep rgl dependencies
+################################################################################
+FROM $BASE_IMAGE as prepper
 ARG DEBIAN_FRONTEND=noninteractive
 
-FROM ${BASE_IMAGE} as rgl-core
-RUN apt update
-RUN apt install -y \
-    git \
-    cmake \
-    python3
+# Edit apt config for caching and update once
+RUN mv /etc/apt/apt.conf.d/docker-clean /etc/apt/ && \
+    echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' \
+      > /etc/apt/apt.conf.d/keep-cache && \
+    apt-get update
+
+# Install bootstrap tools for install scripts
+RUN --mount=type=cache,sharing=locked,target=/var/cache/apt \
+    apt-get install -y --no-install-recommends \
+        cmake \
+        git \
+        python3 \
+        sudo
+
+# Set working directory using standard opt path
+WORKDIR /opt/rgl
+
+# Copy only dependencies definition files
+COPY ./setup.py .
 
-FROM rgl-core AS build
+# install dependencies while caching apt downloads
+# RUN --mount=type=cache,sharing=locked,target=/var/cache/apt \
+#    ./setup.py --install-deps-only
```

Comment:
This `--install-deps-only` arg obviously doesn't exist, but it would be nice if it did.

Even better would be splitting up the setup and compilation scripts, such that changes in the compilation script wouldn't ivalidate the build cache of consecutive layers that could install GB of dependencies.

Or- a package manifest such as a `package.xml` that one could point rosdep at to decouple the dependency lock files from turing complete setup scripts would be even cooler!